connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
To be honest, I'd prefer if you got rid of the Wi-Fi thing in the force-HE patch as well. I think you don't need it. Correct me if i am wrong, but I think the only thing you have to care about is the availability of the cellular network. If it is available, and if it is default, then you are lucky because you don't have to do anything. If it is not default, you force HE. In all other cases you do nothing. For example, I think there is an obvious bug in the current force-HE caused by taking into the equation a dependency which naturally isn't there. For example, if a phone is tethered via Bluetooth or otherwise, your wifi callback wouldn't fire => the app wouldn't behave as devised, would it?